daphne_worker/src/durable/reports_processed.rs

```diff
@@ -178,24 +227,29 @@ impl ReportsProcessed {
                 })
             }
 
-            // Mark reports as aggregated. Return the subset that were already aggregated.
+            // Mark reports as aggregated.
             //
-            // Non-idempotent
+            // If there are any replays, no reports are marked as aggregated.
+            //
+            // Idempotent
             // Input: `Vec<ReportId>`
             // Output: `Vec<ReportId>`
             (DURABLE_REPORTS_PROCESSED_MARK_AGGREGATED, Method::Post) => {
                 let report_ids: Vec<ReportId> = req_parse(&mut req).await?;
-                let replayed_reports = try_join_all(
-                    report_ids
-                        .into_iter()
-                        .map(|report_id| self.to_checked(report_id)),
-                )
-                .await?
-                .into_iter()
-                .flatten()
-                .collect::<Vec<ReportId>>();
+                match self.check_replays(&report_ids).await? {
+                    CheckedReplays::Replayed(replayed) => Response::from_json(&replayed),
+                    CheckedReplays::NotReplayed(not_replayed) => {
```

Comment:
"Replay" is used a lot here. I think using names that are more semantically meaningful here would be helpfulo.
```suggestion
                    CheckReplayResult::SomeReplayed(report_ids) => Response::from_json(&report_ids),
                    CheckReplayResult::AllFresh(report_ids) => {
```